### PR TITLE
[ES DateTimeV2] Fix date periods/ranges inconsistently recognized (#2400)

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
Fix to issue #2400.

The pattern "primera semana" is not currently supported, also in English "first week" is not recognized (but instead "week 1"/"semana 1" is extracted). Should I add support for the pattern?

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs [hunk 1]
```diff
@@ -1895,7 +1895,7 @@ private DateTimeResolutionResult ParseHalfYear(string text, DateObject reference
                 var swift = this.config.GetSwiftYear(orderStr);
                 if (swift < -1)
                 {
-                    return ret;
+                    swift = 0;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs:1898]: Why is this necessary?

File: Specs/DateTime/Spanish/DateTimeModel.json [hunk 2]
```diff
@@ -6714,52 +6702,18 @@
...
-              "timex": "2019-W01",
-              "type": "daterange",
-              "start": "2018-12-31",
-              "end": "2019-01-07"
-            }
-          ]
-        }
-      }
-    ]
+    "Comment": "Also in English, 'first week' is not extracted ('4th week of month/year' and 'week 3' are the supported patterns).",
```
Comment (tellarin) [inline Specs/DateTime/Spanish/DateTimeModel.json:6705]: That makes sense, but wouldn't it be better for "semana" to be extracted by itself in "primera semana" as a duration?
Same in English.

**Review by tellarin — APPROVED** [review]
Approving as-is. In case we decide to implement the extraction of "semana" from input like "1a semana", that can come in a follow up PR.

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs (hunk 1 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs:1898]: otherwise the extraction is parsed only if it contains a year, but this is not always the case e.g. "2º semestre traerá desafíos" (the same behaviour was already implemented in ParseQuarter).

File: Specs/DateTime/Spanish/DateTimeModel.json (hunk 2 above)
Comment (LionbridgeCSII) [reply Specs/DateTime/Spanish/DateTimeModel.json:6705]: do you mean to extract a unit like "week" as duration in case it is preceded by an ordinal number?

Comment (tellarin) [reply Specs/DateTime/Spanish/DateTimeModel.json:6705]: Yes. We can create an issue for it. It's lower priorirty than some of the other fixes.

Comment (tellarin) [reply Specs/DateTime/Spanish/DateTimeModel.json:6705]: Do you see any potential problem with that extraction?

Comment (LionbridgeCSII) [reply Specs/DateTime/Spanish/DateTimeModel.json:6705]: ok, I will open the issue. No I do not see any problem at the moment.
